src/System.Runtime/tests/System/LazyTests.regression.cs

```diff
@@ -0,0 +1,334 @@
+﻿using System.Reflection;
+using System.Threading;
+using Xunit;
+
+namespace System
+{
+	/// <summary>
+    /// This regression suite was designed for testing of a replacement for the original Lazy implementation.
+    /// It does contain some overlap with the existing tests, but I feel that it is still of worth to keep
+    /// the original suite as well as this one. These tests were written with the express purpose of testing
+    /// that the implementation conforms to the existing implementation, rather than to specification.
+    /// The two reasons obviously have major overlap (obviously), but leads to the flavour of this suite 
+    /// being rather dense and repeditive, which for regression I feel is suitable.
```

Comment:
While I appreciate the sentiment here, in a month or a year or whatever, it's not really going to matter that the implementation changed: then we really just want a suite that validates Lazy's behavior.  We should avoid duplication and have a single suite that validates the behavior.